### PR TITLE
Update jib-gradle task to be chains compliant

### LINKED ISSUE #946: Update jib-grade task to work with Chains
# Expected Behavior
`jib-gradle` task should be updated to include `IMAGE_URL` in its results:
```yaml
  results:
  - name: IMAGE_DIGEST
    description: Digest of the image just built.
  - name: IMAGE_URL
    description: URL of the image just built.
```
This will allow Chains to sign and add an attestation to the image by `jib-gradle`

# Actual Behavior
Currently the `jib-gradle` task in the catalog does not work with Chains and the image created is not signed by Chains.

# Steps to Reproduce the Problem

1. Running the current `jib-gradle` task with Chains, the image created is not signed by Chains

### PR DESCRIPTION
**Pull request by bradbeck**
# Changes

Updated `jib-gradle` task to be chains compliant. Fixes #946

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: task/jib-gradle/0.4/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+- **INSECUREREGISTRY**: Whether to allow insecure registry. (*default: "false"*)
+- **EXTRA_ARGS**: Provides extra arguments to the gradle jib build (*default: ""*)
+
+## Workspaces
+
+- **source**: A [Workspace](https://github.com/tektoncd/pipeline/blob/main/docs/workspaces.md) containing the source to build.
+
+## Results
+
+- **IMAGE_DIGEST**: The digest of the image just built.
```
Comment (pxp928) [inline task/jib-gradle/0.4/README.md:28]: nit: IMAGE_URL is missing in results in the Readme

File: task/jib-gradle/0.4/tests/run.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      steps:
+      - name: bash
+        image: ubuntu
+        script: |
+          echo $(params.digest)
+          case .$(params.digest) in
+            ".sha"*) exit 0 ;;
+            *)       echo "Digest value is not correct" && exit 1 ;;
+          esac
+  - name: verify-url
```
Comment (vinamra28) [inline task/jib-gradle/0.4/tests/run.yaml:54]: (not a blocker) can we merge both these tasks ?

File: task/jib-gradle/0.4/jib-gradle.yaml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      subPath: m2-cache
+    - name: $(params.CACHE)
+      mountPath: /workspace/.cache
+      subPath: jib-cache
+    securityContext:
+      runAsUser: 0
+
+  - name: write-url
+    image: $(params.BUILDER_IMAGE)
+    script: echo $(params.IMAGE) | tee $(results.IMAGE_URL.path)
```
Comment (vinamra28) [inline task/jib-gradle/0.4/jib-gradle.yaml:106]: can we merge this step with the previous one as both the steps are using the same image and in step 1 we are already writing the script so maybe we can merge these. WDYT?

**Review by vinamra28 — COMMENTED** [review]
/ok-to-test

File: task/jib-gradle/0.4/tests/run.yaml (hunk 2 above)
Comment (bradbeck) [reply task/jib-gradle/0.4/tests/run.yaml]: 319ef3f6bbbe93792096343a0da9dd90337a3d11

File: task/jib-gradle/0.4/README.md (hunk 1 above)
Comment (bradbeck) [reply task/jib-gradle/0.4/README.md:28]: 5c13eeb3c7709e6b57310c166a08e0f7a2ec022b

**Review by vinamra28 — COMMENTED** [review]
/lgtm
can you please squash the commits as well?

**Comment by bradbeck** [comment]
> /lgtm
> can you please squash the commits as well?

All the commits? Or just the ones after the copy?
I was under the impression that commits were squashed on merge.

**Comment by chmouel** [comment]
/lgtm
/approve

🙇🏻 thank you!

**Pull request merged**